pkg/mapper/mapper_cache.go

```diff
@@ -27,6 +29,18 @@ var (
 	)
 )
 
+type cacheOptions struct {
+	cacheType string
+}
+
+type CacheOption func(*cacheOptions)
```

Comment:
what is the benefit of this construction over, say, passing explicit separate parameters to the constructor?

Reply:
By passing it in as optional arguments, then callers do not need to change their calls.  Within statsd_exporter it's not a big deal to change all the call sites, but I know the mapper package is used by a few other projects, so they could continue to work when they updated their dependencies.